src/main/java/mate/academy/controller/OrderController.java

```diff
@@ -0,0 +1,65 @@
+package mate.academy.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.dto.order.OrderDto;
+import mate.academy.dto.order.item.OrderItemDto;
+import mate.academy.mapper.OrderMapper;
+import mate.academy.model.Status;
+import mate.academy.service.OrderService;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Order management", description = "Endpoints for managing orders")
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/api/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final OrderMapper orderMapper;
+
+    @PreAuthorize("hasRole('USER')")
+    @Operation(summary = "Place an order")
+    @PostMapping
+    public OrderDto placeAnOrder(Long userId, OrderDto orderDto) {
```

Comment:
You should use `@RequestBody` annotation for the object passed to this method. Don't use `userId` it should be taken for example from the user session, but not passed in POST request.
```suggestion
    public OrderDto createOrder(@RequestBody OrderRequestDto request) {
```